src/context/mod.rs

```diff
@@ -105,14 +105,16 @@ pub enum CallKind {
 /// Types of JS values that support the `Borrow` and `BorrowMut` traits can be inspected while the engine is locked by passing a reference to a `Lock` to their methods.
 pub struct Lock<'a> {
     pub(crate) ledger: RefCell<Ledger>,
+    pub(crate) env: Env,
```

Comment:
Could this hold a `Context<'a>` instead or do we rely on the `Env` aliasing?

Reply:
I tried that first and it was difficult because Lock, Ref, RefMut, and the methods on Borrow and BorrowMut then all become generic over the Context trait. But, I think it might still be worth exploring, since on second thought this is a situation where I'm using the lifetime-less Env to work around the borrow checker. I actually specifically ran into compile errors where lifetimes of `Context::lock()` and `Context::borrow()` mismatched … Adding a lifetime parameter to Env would likely result in the same kind of problem. I didn't fully understand those errors though. I'll push that on a separate branch in a bit

Reply:
this is where I got more or less: https://github.com/goto-bus-stop/neon/commit/b1cf3170ccc87f27e20740f632addb2359a888f9